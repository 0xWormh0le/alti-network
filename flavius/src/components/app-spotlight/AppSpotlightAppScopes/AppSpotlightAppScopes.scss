.AppSpotlightAppScopes {
  line-height: 26px;
  letter-spacing: normal;
  font-size: 14px;
  color: #6e6e6e;
  height: 84px;
  padding-inline-start: 20px;
  position: absolute;
  width: 100%;
  background-color: white;
  z-index: 10;
  height: fit-content;
}
