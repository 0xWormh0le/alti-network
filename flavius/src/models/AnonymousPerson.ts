import Person from './Person'
import { UsageKind } from 'types/common'

const email: Email = {
  address: 'sample@foo.bar',
  kind: UsageKind.personal,
  primary: false,
  accessCount: 1,
  riskCount: 1,
  lastDeletedPermissions: 0
}

class AnonymousPerson extends Person {
  constructor() {
    super({
      name: {
        familyName: 'User',
        givenName: 'Anonymous',
        fullName: 'User Anonymous'
      },
      primaryEmail: email
    } as IPerson)
  }
}

export default AnonymousPerson
