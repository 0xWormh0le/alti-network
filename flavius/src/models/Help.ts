import UI_STRINGS from 'util/ui-strings'

export interface IHelpQuestion {
  question: string
  risk_ids: RiskTypeId[]
}

export const HelpQuestionsList: IHelpQuestion[] = [
  {
    question: UI_STRINGS.HELP.SHARING_RISKS,
    risk_ids: [1011, 1013, 1020, 1021, 1022, 1023]
  },
  {
    question: UI_STRINGS.HELP.RELATIONSHIP_RISKS,
    risk_ids: [2000, 2001, 2002, 2003]
  },
  {
    question: UI_STRINGS.HELP.ACTIVITY_RISKS,
    risk_ids: [3100, 3200, 3010, 3012]
  },
  {
    question: UI_STRINGS.HELP.INFORMATIONAL_RISKS,
    risk_ids: [0, 10, 11]
  }
]
