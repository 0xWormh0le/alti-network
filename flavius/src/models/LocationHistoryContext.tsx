import React from 'react'
import pick from 'lodash/pick'
import { noop } from 'lodash'

const TRACKED_PARAMS = [
  'page',
  'pageNumber',
  'riskTypeIds',
  'selectedCardKey',
  'selectedSubNavKey',
  'selectedEmail',
  'platformIds',
  'userVisibilityState'
]

export const LocationHistoryContext = React.createContext({} as LocationHistoryContextProviderState)

interface LocationHistoryContextProviderProps {
  children: React.ReactNode
}

interface LocationHistoryContextProviderState {
  breadcrumbLocations: string[]
  trackedQueryParams: {}
  deleteFromBreadcrumbLocations?: (path: string) => void
  onLocationVisited?: (location: string, breadcrumbPosition?: number) => void
  onQueryParamsChanged?: (queryParams: object) => void
}

export class LocationHistoryContextProvider extends React.Component<
  LocationHistoryContextProviderProps,
  LocationHistoryContextProviderState
> {
  public state = {
    breadcrumbLocations: [], // array of the last location visited
    trackedQueryParams: {}, // keep track of the latest value for some interesting query params before opening modals
    deleteFromBreadcrumbLocations: noop
  }

  public render() {
    return (
      <LocationHistoryContext.Provider
        value={{
          breadcrumbLocations: this.state.breadcrumbLocations,
          deleteFromBreadcrumbLocations: (path) => {
            this.setState((prevState) => {
              return {
                breadcrumbLocations: prevState.breadcrumbLocations.slice(0, prevState.breadcrumbLocations.indexOf(path))
              }
            })
          },
          onLocationVisited: (location: string, breadcrumbPosition?: number) => {
            if (breadcrumbPosition === undefined) {
              // push new location to the back of the array and just keep the last 3
              this.setState((prevState) => ({
                breadcrumbLocations: [...prevState.breadcrumbLocations, location].slice(-3)
              }))
            } else {
              // this means the user clicked on the breadcrumb and navigated to a breadcrumb path,
              // we need to reset the locations stack in that case
              this.setState((prevState) => ({
                breadcrumbLocations: prevState.breadcrumbLocations.slice(0, breadcrumbPosition + 1)
              }))
            }
          },
          trackedQueryParams: this.state.trackedQueryParams,
          onQueryParamsChanged: (queryParams: object) => {
            this.setState({
              trackedQueryParams: pick(queryParams, TRACKED_PARAMS)
            })
          }
        }}>
        {this.props.children}
      </LocationHistoryContext.Provider>
    )
  }
}

// Consumer
export const LocationHistoryContextConsumer = LocationHistoryContext.Consumer
