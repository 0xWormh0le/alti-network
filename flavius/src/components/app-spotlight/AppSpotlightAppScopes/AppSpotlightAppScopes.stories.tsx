import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import AppSpotlightAppScopes, { AppSpotlightAppScopesProps } from './AppSpotlightAppScopes'

export default {
  title: 'AppSpotlight/AppSpotlightAppScopes',
  component: AppSpotlightAppScopes
} as Meta

const Template: Story<AppSpotlightAppScopesProps> = (args) => <AppSpotlightAppScopes {...args} />

export const Default = Template.bind({})
Default.args = {
  scopes: ['Scope1', 'Scope2']
}
