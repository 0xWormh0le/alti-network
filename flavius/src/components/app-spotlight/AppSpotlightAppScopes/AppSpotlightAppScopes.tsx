import React from 'react'

import './AppSpotlightAppScopes.scss'

export interface AppSpotlightAppScopesProps {
  scopes: string[]
}

export const AppSpotlightAppScopes: React.SFC<AppSpotlightAppScopesProps> = ({ scopes }) => {
  return (
    <ul className='AppSpotlightAppScopes'>
      {scopes?.map((scope, index) => (
        <li key={index.toString()}>{scope}</li>
      ))}
    </ul>
  )
}

export default AppSpotlightAppScopes
