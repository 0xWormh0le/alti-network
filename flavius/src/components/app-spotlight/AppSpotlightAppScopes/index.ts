import { AppSpotlightAppScopes } from './AppSpotlightAppScopes'

export default AppSpotlightAppScopes
