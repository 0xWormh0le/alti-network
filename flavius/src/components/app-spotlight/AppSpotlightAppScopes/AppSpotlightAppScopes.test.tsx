import React from 'react'

import { renderWithRouter } from 'test/support/helpers'
import AppSpotlightAppScopes from './AppSpotlightAppScopes'
import { ScopeCatalog } from 'models/ScopeCatalog'

describe('AppSpotlightAppScopes', () => {
  it('renders correctly', async () => {
    const scopes = Object.keys(ScopeCatalog)
    const { container } = renderWithRouter(<AppSpotlightAppScopes scopes={scopes} />)
    expect(container).toMatchSnapshot()
  })
})

describe('Grant access behavior', () => {
  it('changes content per item selection', async () => {
    const scopes = Object.keys(ScopeCatalog)
    const { container } = renderWithRouter(<AppSpotlightAppScopes scopes={scopes} />)

    const firstGrant = container.querySelectorAll('li')[0]

    expect(firstGrant).toHaveTextContent(scopes[0])
  })
})
