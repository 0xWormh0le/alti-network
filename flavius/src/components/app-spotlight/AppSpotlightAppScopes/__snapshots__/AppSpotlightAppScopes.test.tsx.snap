// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSpotlightAppScopes renders correctly 1`] = `
<div>
  <ul
    class="AppSpotlightAppScopes"
  >
    <li>
      openid
    </li>
    <li>
      https://www.googleapis.com/auth/youtubepartner
    </li>
    <li>
      https://www.googleapis.com/auth/youtube.upload
    </li>
    <li>
      https://www.googleapis.com/auth/youtube.tve
    </li>
    <li>
      https://www.googleapis.com/auth/youtube-paid-content
    </li>
    <li>
      https://www.googleapis.com/auth/youtube
    </li>
    <li>
      https://www.googleapis.com/auth/userinfo.profile
    </li>
    <li>
      https://www.googleapis.com/auth/userinfo.email
    </li>
    <li>
      https://www.googleapis.com/auth/userinfo#email
    </li>
    <li>
      https://www.googleapis.com/auth/user.phonenumbers.read
    </li>
    <li>
      https://www.googleapis.com/auth/user.emails.read
    </li>
    <li>
      https://www.googleapis.com/auth/user.birthday.read
    </li>
    <li>
      https://www.googleapis.com/auth/urlshortener
    </li>
    <li>
      https://www.googleapis.com/auth/tasks
    </li>
    <li>
      https://www.googleapis.com/auth/sqlservice.admin
    </li>
    <li>
      https://www.googleapis.com/auth/sqlservice
    </li>
    <li>
      https://www.googleapis.com/auth/spreadsheets.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/spreadsheets.currentonly
    </li>
    <li>
      https://www.googleapis.com/auth/spreadsheets
    </li>
    <li>
      https://www.googleapis.com/auth/siteverification
    </li>
    <li>
      https://www.googleapis.com/auth/script.webapp.deploy
    </li>
    <li>
      https://www.googleapis.com/auth/script.storage
    </li>
    <li>
      https://www.googleapis.com/auth/script.send_mail
    </li>
    <li>
      https://www.googleapis.com/auth/script.scriptapp
    </li>
    <li>
      https://www.googleapis.com/auth/script.locale
    </li>
    <li>
      https://www.googleapis.com/auth/script.external_request
    </li>
    <li>
      https://www.googleapis.com/auth/script.container.ui
    </li>
    <li>
      https://www.googleapis.com/auth/pubsub
    </li>
    <li>
      https://www.googleapis.com/auth/profile.emails.read
    </li>
    <li>
      https://www.googleapis.com/auth/presentations.currentonly
    </li>
    <li>
      https://www.googleapis.com/auth/presentations
    </li>
    <li>
      https://www.googleapis.com/auth/plus.profile.language.read
    </li>
    <li>
      https://www.googleapis.com/auth/plus.profile.emails.read
    </li>
    <li>
      https://www.googleapis.com/auth/plus.profile.agerange.read
    </li>
    <li>
      https://www.googleapis.com/auth/plus.me
    </li>
    <li>
      https://www.googleapis.com/auth/photoslibrary.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/photos
    </li>
    <li>
      https://www.googleapis.com/auth/peopleapi.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/payments.video_purchase
    </li>
    <li>
      https://www.googleapis.com/auth/payments.make_payments
    </li>
    <li>
      https://www.googleapis.com/auth/monitoring.write
    </li>
    <li>
      https://www.googleapis.com/auth/monitoring.read
    </li>
    <li>
      https://www.googleapis.com/auth/monitoring
    </li>
    <li>
      https://www.googleapis.com/auth/meetings
    </li>
    <li>
      https://www.googleapis.com/auth/logging.write
    </li>
    <li>
      https://www.googleapis.com/auth/logging.read
    </li>
    <li>
      https://www.googleapis.com/auth/logging.admin
    </li>
    <li>
      https://www.googleapis.com/auth/iam.test
    </li>
    <li>
      https://www.googleapis.com/auth/hangout.participants
    </li>
    <li>
      https://www.googleapis.com/auth/hangout.av
    </li>
    <li>
      https://www.googleapis.com/auth/groups
    </li>
    <li>
      https://www.googleapis.com/auth/googletalk
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.settings.sharing
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.settings.basic
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.send
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.modify
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.labels
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.insert
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.compose
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.addons.execute
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.addons.current.message.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.addons.current.message.metadata
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.addons.current.message.action
    </li>
    <li>
      https://www.googleapis.com/auth/gmail.addons.current.action.compose
    </li>
    <li>
      https://www.googleapis.com/auth/gerritcodereview
    </li>
    <li>
      https://www.googleapis.com/auth/forms.currentonly
    </li>
    <li>
      https://www.googleapis.com/auth/forms
    </li>
    <li>
      https://www.googleapis.com/auth/flexible-api
    </li>
    <li>
      https://www.googleapis.com/auth/firebase.messaging
    </li>
    <li>
      https://www.googleapis.com/auth/firebase
    </li>
    <li>
      https://www.googleapis.com/auth/ediscovery
    </li>
    <li>
      https://www.googleapis.com/auth/drive.readonly.metadata
    </li>
    <li>
      https://www.googleapis.com/auth/drive.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/drive.photos.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/drive.metadata.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/drive.metadata
    </li>
    <li>
      https://www.googleapis.com/auth/drive.install
    </li>
    <li>
      https://www.googleapis.com/auth/drive.file
    </li>
    <li>
      https://www.googleapis.com/auth/drive.apps.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/drive.appfolder
    </li>
    <li>
      https://www.googleapis.com/auth/drive.appdata
    </li>
    <li>
      https://www.googleapis.com/auth/drive.activity
    </li>
    <li>
      https://www.googleapis.com/auth/drive
    </li>
    <li>
      https://www.googleapis.com/auth/documents.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/documents.currentonly
    </li>
    <li>
      https://www.googleapis.com/auth/documents
    </li>
    <li>
      https://www.googleapis.com/auth/docs.test
    </li>
    <li>
      https://www.googleapis.com/auth/docs
    </li>
    <li>
      https://www.googleapis.com/auth/devstorage.read_write
    </li>
    <li>
      https://www.googleapis.com/auth/devstorage.read_only
    </li>
    <li>
      https://www.googleapis.com/auth/devstorage.full_control
    </li>
    <li>
      https://www.googleapis.com/auth/contactstore.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/contacts.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/contacts
    </li>
    <li>
      https://www.googleapis.com/auth/compute.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/compute
    </li>
    <li>
      https://www.googleapis.com/auth/codejam
    </li>
    <li>
      https://www.googleapis.com/auth/cloudprint
    </li>
    <li>
      https://www.googleapis.com/auth/cloud-platform.read-only
    </li>
    <li>
      https://www.googleapis.com/auth/cloud-platform
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.topics
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.rosters
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.profile.photos
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.profile.emails
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.guardianlinks.students
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.coursework.students
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.coursework.me
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.courses
    </li>
    <li>
      https://www.googleapis.com/auth/classroom.announcements
    </li>
    <li>
      https://www.googleapis.com/auth/chromewebstore
    </li>
    <li>
      https://www.googleapis.com/auth/carpool
    </li>
    <li>
      https://www.googleapis.com/auth/carddav
    </li>
    <li>
      https://www.googleapis.com/auth/calendar.settings.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/calendar.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/calendar.events.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/calendar.events
    </li>
    <li>
      https://www.googleapis.com/auth/calendar.addons.execute
    </li>
    <li>
      https://www.googleapis.com/auth/calendar
    </li>
    <li>
      https://www.googleapis.com/auth/bigquery.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/bigquery.insertdata
    </li>
    <li>
      https://www.googleapis.com/auth/bigquery
    </li>
    <li>
      https://www.googleapis.com/auth/apps.order
    </li>
    <li>
      https://www.googleapis.com/auth/apps.licensing
    </li>
    <li>
      https://www.googleapis.com/auth/apps.groups.settings
    </li>
    <li>
      https://www.googleapis.com/auth/apps.groups.migration
    </li>
    <li>
      https://www.googleapis.com/auth/android_video
    </li>
    <li>
      https://www.googleapis.com/auth/analytics.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/analytics.edit
    </li>
    <li>
      https://www.googleapis.com/auth/admin.reports.usage.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.reports.audit.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.userschema.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.userschema
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.user.security
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.user.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.user.alias
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.user
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.rolemanagement
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.resource.calendar.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.resource.calendar
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.orgunit
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.notifications
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.group.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.group.member.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.group.member
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.group
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.domain.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.domain
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.device.mobile.action
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.device.mobile
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.device.chromeos.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.device.chromeos
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.customer.readonly
    </li>
    <li>
      https://www.googleapis.com/auth/admin.directory.customer
    </li>
    <li>
      https://www.googleapis.com/auth/admin.datatransfer
    </li>
    <li>
      https://www.googleapis.com/auth/activity
    </li>
    <li>
      https://www.googleapis.com/auth/actions.builder
    </li>
    <li>
      https://www.googleapis.com/auth/accounts.reauth
    </li>
    <li>
      https://www.google.com/m8/feeds
    </li>
    <li>
      https://www.google.com/hosted/services/v1.0/reports/ReportingData
    </li>
    <li>
      https://www.google.com/calendar/feeds/
    </li>
    <li>
      https://www.google.com/analytics/feeds/
    </li>
    <li>
      https://www.google.com/accounts/OAuthLogin
    </li>
    <li>
      https://spreadsheets.google.com/feeds/
    </li>
    <li>
      https://sites.google.com/feeds
    </li>
    <li>
      https://picasaweb.google.com/data/
    </li>
    <li>
      https://photos.googleapis.com/data/
    </li>
    <li>
      https://mail.google.com/
    </li>
    <li>
      https://docs.googleusercontent.com/
    </li>
    <li>
      https://docs.google.com/feeds/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/user/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/policies/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/groups/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/emailsettings/2.0/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/domain/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/compliance/audit/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/calendar/resource/
    </li>
    <li>
      https://apps-apis.google.com/a/feeds/alias/
    </li>
  </ul>
</div>
`;
